collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java

```diff
@@ -355,8 +358,23 @@ public PropertyManager providesPropertyManager(Application application, RxEventB
     }
 
     @Provides
-    public SettingsChangeHandler providesSettingsChangeHandler(PropertyManager propertyManager, FormUpdateManager formUpdateManager) {
-        return new CollectSettingsChangeHandler(propertyManager, formUpdateManager);
+    public ServerRepository providesServerRepository() {
+        return new ServerRepository() {
```

Comment:
We can follow up with an actual tested implementation (replacing `ChangingServerUrlUtils`)